posts/2022-03-17-graphql-22004-beta.adoc

```diff
@@ -0,0 +1,327 @@
+---
+layout: post
+title: "MicroProfile GraphQL 2.0, Java 18 support, and distributed security caches on Open Liberty 22.0.0.4-beta"
+# Do NOT change the categories section
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/ryan-storey
+author_github: https://github.com/ryan-storey
+seo-title: MicroProfile GraphQL 2.0, Java 18 support, and distributed security caches on Open Liberty 22.0.0.4-beta - OpenLiberty.io
+seo-description: Open Liberty 22.0.0.4-beta offers GraphQL 2.0, incorporating Jakarta EE 9.1 dependencies, as well as Java 18 support with many new exciting features and changes. Also included in this release is distributed security caches, allowing multiple Liberty servers to share caches via a JCache provider.
+blog_description: Open Liberty 22.0.0.4-beta offers GraphQL 2.0, incorporating Jakarta EE 9.1 dependencies, as well as Java 18 support with many new exciting features and changes. Also included in this release is distributed security caches, allowing multiple Liberty servers to share caches via a JCache provider.
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+---
+= MicroProfile GraphQL 2.0, Java 18 support, and distributed security caches on Open Liberty 22.0.0.4-beta
+Ryan Storey <https://github.com/ryan-storey>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+//Blank line here is necessary before starting the body of the post.
+
+Open Liberty 22.0.0.4-beta offers GraphQL 2.0, incorporating Jakarta EE 9.1 dependencies, as well as Java 18 support with many new exciting features and changes. Also included in this release is distributed security caches, allowing multiple Liberty servers to share caches via a JCache provider.
+
+The link:{url-about}[Open Liberty] 22.0.0.4-beta includes the following beta features (along with all GA features):
+
+* <<graphql, MicroProfile GraphQL 2.0>>
+* <<java18, Java 18 Support>>
+* <<security_caches, Distributed Security Caches>>
+
+[#graphql]
+== MicroProfile GraphQL 2.0
+
+MicroProfile GraphQL 2.0 incorporates Jakarta EE 9.1 dependencies.  This allows developers to continue using the same functionality provided by MP GraphQL 1.0 but with the updated Jakarta components such as link:{url-prefix}/docs/latest/reference/feature/cdi-3.0.html[CDI 3.0], link:{url-prefix}/docs/latest/reference/feature/restfulWS-3.0.html[restfulWS] 3.0, link:{url-prefix}/docs/latest/reference/feature/jsonb-2.0.html[JSON-B 2.0], etc. Functionally, version 2.0 behaves the same as 1.0 because it uses the same version of the underlying implementation (link:https://github.com/smallrye/smallrye-graphql[SmallRye GraphQL]).  If you'd like to learn more about MP GraphQL, see our previous link:{url-prefix}/blog/2020/06/10/microprofile-graphql-open-liberty.html[blog post about MP GraphQL 1.0] and our link:{url-prefix}/blog/2020/08/28/graphql-apis-open-liberty-20009.html[blog post on how to use the GraphiQL UI and third-party Client APIs].
```

Comment:
```suggestion
MicroProfile GraphQL 2.0 incorporates Jakarta EE 9.1 dependencies.  This allows developers to continue using the same functionality provided by MP GraphQL 1.0 but with the updated Jakarta features such as link:{url-prefix}/docs/latest/reference/feature/cdi-3.0.html[CDI 3.0], link:{url-prefix}/docs/latest/reference/feature/restfulWS-3.0.html[restfulWS] 3.0, link:{url-prefix}/docs/latest/reference/feature/jsonb-2.0.html[JSON-B 2.0], etc. Functionally, version 2.0 behaves the same as 1.0 because it uses the same version of the underlying implementation (link:https://github.com/smallrye/smallrye-graphql[SmallRye GraphQL]).  If you'd like to learn more about MP GraphQL, see our previous link:{url-prefix}/blog/2020/06/10/microprofile-graphql-open-liberty.html[blog post about MP GraphQL 1.0] and our link:{url-prefix}/blog/2020/08/28/graphql-apis-open-liberty-20009.html[blog post on how to use the GraphiQL UI and third-party Client APIs].
```